wrap content in IndexableObjectWrapper for indexing PDFs

huubbouma opened the pull request:
Hi there,

This patch uses the standard plone way of indexing the content, instead of directly calling the SearchableText method. This solves my issue of indexing PDF files.

Cheers, Huub

huubbouma committed:
wrap content in IndexableObjectWrapper for indexing PDFs

gotcha commented:
Would you mind filing an issue where you describe what you mean with your issue of indexing PDF files ?

huubbouma commented:
Hi @gotcha I just filed issue #6

Commit:
Added state change event handler

vincentfretin commented on src/collective/elasticindex/changes.py:
```diff
@@ -71,7 +72,15 @@ def get_data(content, security=False, domain=None):
+        # delegated to IIndexer adapters as appropriate.
+        # This will e.g. take care of PDF file indexing.
+        cat = getToolByName(content, 'portal_catalog')
+        wrapper = IndexableObjectWrapper(content, cat)
```
Instead of hard coding the wrapper, retrieve it with queryMultiAdapter like it is done in Products/CMFPlone/CatalogTool.py:catalog_object
code not tested:

```
from plone.indexer.interfaces import IIndexableObject
cat = getToolByName(content, 'portal_catalog')
wrapper = content                                                              
if not IIndexableObject.providedBy(content):                             
    # This is the CMF 2.2 compatible approach, which should be used        
    # going forward                                                     
    w = queryMultiAdapter((content, cat), IIndexableObject)          
    if wrapper is not None:                                             
        wrapper = w
    text = getattr(wrapper, 'SearchableText', None)
    if safe_callable(text):
        text = text()
```

kcleong committed:
Update mapping to support ES 6.x

kcleong committed:
Add support for search auto suggest

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters-completion.html

kcleong committed:
Enable Dutch language support when this language is active
